targets/ingester/src/transform/fichesServicePublic/parseReference.ts

```diff
@@ -35,85 +35,70 @@ const getTextType = (qs) => {
   return null;
 };
 
-/**
- *
- * @param {string} url
- */
-function isPreviousLegifranceUrl(url) {
+function isPreviousLegifranceUrl(url: string) {
   return /affich.+\.do/.test(url);
 }
 
-/**
- *
- * @param {import("@socialgouv/legi-data-types").CodeArticle} article
- */
-function cdtArticleReference(article) {
+function cdtArticleReference(
+  article: CodeArticle
+): ServicePublicInternalReference {
   return {
     slug: slugify(fixArticleNum(article.data.id, article.data.num)),
     title: article.data.num,
     type: SOURCES.CDT,
   };
 }
-/**
- *
- * @param {import("@socialgouv/kali-data-types").IndexedAgreement} convention
- */
-function agreementReference(convention) {
-  const { num, shortTitle } = convention;
+
+function agreementReference(
+  agreement: IndexedAgreement
+): ServicePublicInternalReference {
+  const { num, shortTitle } = agreement;
   return {
     slug: slugify(`${num}-${shortTitle}`.substring(0, 80)),
     title: shortTitle,
     type: SOURCES.CCN,
   };
 }
-/**
- *
- * @param {string} url
- * @param {string} label
- */
-function externalReference(url, label) {
+
+function externalReference(
+  url: string,
+  label: string
+): ServicePublicExternalReference {
   return {
     title: label,
     type: SOURCES.EXTERNALS,
     url,
   };
 }
 
-/**
- * @param {import("@socialgouv/fiches-vdd-types").RawJson[]} references
- * @param {ingester.referenceResolver} resolveCdtReference
- * @param {import("@socialgouv/kali-data-types").IndexedAgreement[]} agreements
- * @returns { ingester.ReferencedTexts[] }
- */
-export function parseReferences(references, resolveCdtReference, agreements) {
-  /** @type {ingester.ReferencedTexts[][]} */
+export function parseReferences(
+  references: RawJson[],
+  resolveCdtReference: ReferenceResolver,
+  agreements: IndexedAgreement[]
+): ServicePublicReference[] {
   const referencedTexts = [];
 
   for (const reference of references) {
     const { URL: url } = reference.attributes;
-    const label = reference.children[0].children[0].text;
+    const label = reference.children?.[0]?.children?.[0]?.name;
```

Comment:
Same here, `children` can't be null or undefined from the definition of `@socialgouv/fiches-vdd-types`. The optional chaining is useless or definition is wrong.